From 80f6c7e47f564456453eeb3a5f7f10db4735fe67 Mon Sep 17 00:00:00 2001
From: Sridharan S N <quic_sridsn@quicinc.com>
Date: Fri, 1 Dec 2023 11:39:28 +0530
Subject: [PATCH 398/500] drivers: soc: qcom: Add support for various drivers.

Add support for:
QCOM_DCC_V2
IPQ_DEBUG
MSM_RPM_LOG
IPQ_REG_UPDATE
QTI_TZ_LOG
QTI_LICENSE_MANAGER

Change-Id: I58da44d6218a0105c1c72cc79af94fc26f854da7
Signed-off-by: Sridharan S N <quic_sridsn@quicinc.com>
---
 drivers/soc/qcom/Kconfig  | 43 +++++++++++++++++++++++++++++++++++++++
 drivers/soc/qcom/Makefile |  6 ++++++
 2 files changed, 49 insertions(+)

diff --git a/drivers/soc/qcom/Kconfig b/drivers/soc/qcom/Kconfig
index 6916ac5bd2be..e31bc2a778af 100644
--- a/drivers/soc/qcom/Kconfig
+++ b/drivers/soc/qcom/Kconfig
@@ -301,4 +301,47 @@ config QTI_EMULATION_APSS_ACC
 	  to the ACC associated with the CPU accessing the region.
 
 	  If unsure, say N.
+
+config QCOM_DCC_V2
+	bool "QCOM Data Capture and Compare enigne support"
+	help
+	  This option enables driver for Data Capture and Compare engine. DCC
+	  V2 driver provides interface to configure DCC block and read back
+	  captured data from DCC_V2's internal SRAM
+
+config IPQ_DEBUG
+	bool "QTI, Subsystem IPQ debug driver"
+	depends on ARCH_QCOM
+	help
+	 Generic driver to support debug features
+
+config MSM_RPM_LOG
+	tristate "QTI Resource Power Manager Log Driver"
+	help
+	 This option enables a driver which can read from a circular buffer
+	 of messages produced by the RPM. These messages provide diagnostic
+	 information about RPM operation. The driver outputs the messages
+	 via a debugfs node.
+
+config IPQ_REG_UPDATE
+	bool "QTI, Subsystem register update driver"
+	depends on ARCH_QCOM
+	help
+	 Generic driver to configure required registers
+
+config QTI_TZ_LOG
+	tristate "Trust Zone (TZ) Log Driver"
+	depends on DEBUG_FS
+	help
+	  This option enables a driver with a debugfs interface for messages
+	  produced by the Secure code (Trust zone). These messages provide
+	  diagnostic information about TZ operation.
+
+config QTI_LICENSE_MANAGER
+	tristate "QTI License manager driver"
+	help
+	  This option enables a driver with License Manager QMI Service
+	  running to communicate the license termination mode and license
+	  files to the attached PCIe device.
+
 endmenu
diff --git a/drivers/soc/qcom/Makefile b/drivers/soc/qcom/Makefile
index a6dbbc38a3fc..b8cd29050acf 100644
--- a/drivers/soc/qcom/Makefile
+++ b/drivers/soc/qcom/Makefile
@@ -33,3 +33,9 @@ obj-$(CONFIG_QCOM_ICC_BWMON)	+= icc-bwmon.o
 qcom_ice-objs			+= ice.o
 obj-$(CONFIG_QCOM_INLINE_CRYPTO_ENGINE)	+= qcom_ice.o
 obj-$(CONFIG_QTI_EMULATION_APSS_ACC) += cpu_ops.o cpu_pwr_ctl.o
+obj-$(CONFIG_QCOM_DCC_V2)	+= dcc_v2.o
+obj-$(CONFIG_IPQ_DEBUG)	+= ipq-debug.o
+obj-$(CONFIG_MSM_RPM_LOG)	+= rpm_log.o
+obj-$(CONFIG_IPQ_REG_UPDATE) += ipq_reg_update.o
+obj-$(CONFIG_QTI_TZ_LOG) += tz_log.o
+obj-$(CONFIG_QTI_LICENSE_MANAGER) += license_manager.o
-- 
2.34.1

